remote/remote_test.go

```diff
@@ -1500,4 +1500,82 @@ func testImage(t *testing.T, when spec.G, it spec.S) {
 			})
 		})
 	})
+
+	when("#CheckReadAccess", func() {
+		when("image exists in the registry and client has read access", func() {
+			it("returns true", func() {
+				origImage, err := remote.NewImage(repoName, authn.DefaultKeychain)
+				h.AssertNil(t, err)
+				h.AssertNil(t, origImage.Save())
+
+				image, err := remote.NewImage(repoName, authn.DefaultKeychain)
+				h.AssertNil(t, err)
+
+				h.AssertEq(t, image.CheckReadAccess(), true)
+			})
+		})
+
+		when("image does not exist in the registry and client has read access", func() {
+			it("returns true", func() {
+				image, err := remote.NewImage(repoName, authn.DefaultKeychain)
+				h.AssertNil(t, err)
+				h.AssertEq(t, image.CheckReadAccess(), true)
+			})
+		})
+
+		when("image does not exist in the registry and client hasn't read access", func() {
```

Comment:
```suggestion
		when("image does not exist in the registry and client doesn't have read access", func() {
```

Reply:
I addressed this suggestion in the following [commit](https://github.com/buildpacks/imgutil/pull/123/commits/d588cd706db0e32f55858fa00211c948d045210a)